File: apps/api/src/models/MmfFund.ts
import mongoose from 'mongoose';

const mmfFundSchema = new mongoose.Schema({
  name: { type: String, required: true },
  provider: { type: String, required: true },
  interestRate: { type: Number, required: true },
  minimumInvestment: { type: Number, required: true },
  riskLevel: { type: String, enum: ['low', 'medium', 'high'], default: 'low' },
  maturityDays: { type: Number, default: 0 },
  totalAum: { type: Number, required: true },
  description: { type: String, default: '' },
  websiteUrl: { type: String, default: '' },
  isActive: { type: Boolean, default: true }
}, { timestamps: true });

export const MmfFund = mongoose.model('MmfFund', mmfFundSchema);

File: OneDrive/Desktop/SharedProj/HazinaHub/backend/app/models/mmf_fund.py
from datetime import datetime
from typing import Literal
from beanie import Document
from pydantic import Field


class MmfFund(Document):
    name: str
    provider: str
    interest_rate: float
    minimum_investment: float = 1000.0
    risk_level: Literal["low", "medium", "high", "sovereign"] = "low"
    maturity_days: int = 0
    total_aum: float = 0.0
    description: str = ""
    website_url: str = ""
    logo_url: str = ""
    asset_class: Literal["MMF", "SACCO", "T-Bill", "Stock"] = "MMF"
    is_active: bool = True
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "mmf_funds"
        indexes = ["is_active"]
